### ISSUE TITLE
Still unable to download foreign file from URL with parameters

### ISSUE THREAD

**Issue by mhoban** [nextflow-io/nextflow#4092]
This issue was previously reported in #2918, but I'm still having the problem in 23.06.0-edge build 5865.

### Expected behavior and actual behavior

As in #2918, Channel.fromPath should download a URL to a local file, even when the URL has query parameters (i.e., stuff following the '?'). In addition, the local filename for the downloaded URL can be unexpected. Maybe the addition of a localName option in fromPath could help this?

### Steps to reproduce the problem

here's a simple script that just looks at what results from the Channel.fromPath operation:
```nextflow
#!/usr/bin/env nextflow
nextflow.enable.dsl=2
 
workflow {
    Channel.fromPath('https://www.dropbox.com/s/fv3dpvws6zjvtib/classifier.rds?dl=1') | view
    Channel.fromPath('https://api.duckduckgo.com/?format=json&q=hello') | view
}
```

When I run this, it results in:
```
N E X T F L O W  ~  version 23.06.0-edge
Launching `./f2.nf` [clever_aryabhata] DSL2 - revision: 07d1e95f10
/
/s/fv3dpvws6zjvtib/classifier.rds
```

Since the duckduckgo.com URL doesn't have a set filename, it is stored locally as just '/'. This causes issues. The dropbox URL appears to work properly until you view the contents of the file, and you'll see that it's the HTML page returned by the URL *without* the '?dl=1' part

Here's a short script that shows that behavior:
```nextflow
#!/usr/bin/env nextflow
nextflow.enable.dsl=2
 
process thing {
  debug true

  input: 
    path(f1)
  output: 
    path(f1)

  publishDir "output"

  script:
  """
  echo ${f1}
  """
}


workflow {
    Channel.fromPath('https://www.dropbox.com/s/fv3dpvws6zjvtib/classifier.rds?dl=1') | thing
}
```

output:
```
$ ./f2.nf
N E X T F L O W  ~  version 23.06.0-edge
Launching `./f2.nf` [lonely_woese] DSL2 - revision: 47bae78354
executor >  local (1)
[f7/338cde] process > thing (1) [100%] 1 of 1 ✔
classifier.rds

$ head output/classifier.rds
<!DOCTYPE html>
  <html class="maestro global-header" xmlns="http://www.w3.org/1999/xhtml" lang="en">
    <head>
<meta charset="utf-8" />
<meta content="noindex, nofollow, noimageindex" name="robots" />
<meta content="width=device-width, initial-scale=1" name="viewport" />
<meta content="Shared with Dropbox" property="og:description" />
<meta content="Dropbox" property="og:site_name" />
<meta content="/s/fv3dpvws6zjvtib/classifier.rds" property="og:url" />
<meta content="classifier.rds" property="og:title" />
```

As you can see, the contents are HTML, when they should be binary. If you visit this link in a browser or using curl, you get the correct file.

### Environment 

* Nextflow version: 23.06.0-edge build 5865
* Java version: openjdk version "11.0.17" 2022-10-18
* Operating system: Ubuntu 22.04.1 LTS
* Bash version: GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)

**Comment by mhoban** [comment]
The solution seems to have reverted between 21.10.6 and 21.12.1, at least according to the sequential releases. The main difference from a user perspective is that 21.10.6 treats the '?' as an escape-able glob character, whereas subsequent releases do not.

**Comment by pditommaso** [comment]
TLDR; the example works properly using adding `glob:false` in the fromPath method, eg. 

```
workflow {
    Channel.fromPath('https://www.dropbox.com/s/fv3dpvws6zjvtib/classifier.rds?dl=1') | thing
}
```

#### The long story 

`channel.fromPath` path by default interprets `*` and `?` as glob wildcard, this is the reason it strips the `?dl=1` parameter from the URL. Specifying the option `glob:false` it is handled properly. 


Regarding the snippet 

```
 Channel.fromPath('https://www.dropbox.com/s/fv3dpvws6zjvtib/classifier.rds?dl=1') | view
```

Here there's another tricky behaviour caused by the `Path.toString()` API that's used by nextflow. In a nutshell the `view` operator converts to a string the path object using the `toString` method, however, this only renders the *path* component and therefore strips away the host and the parameter from the URL. 

If you use the following, you will see that's rendered correctly 

```
    Channel.fromPath('https://www.dropbox.com/s/fv3dpvws6zjvtib/classifier.rds?dl=1', glob:false) | view {  it.toUriString() }
```

Said in another way `channel.fromPath` returns `Path` objects, it required to use the method `toUriString()` to map to the correct string version.
